test/xbgpu/test_bsend.py

```diff
@@ -0,0 +1,190 @@
+    """Test :class:`katgpucbf.xbgpu.bsend.BSend`."""
+
+    @staticmethod
+    async def _send_data(
+        send_stream: BSend,
+        heap_timestamp_step: int,
+        time_converter: TimeConverter,
+        sensors: SensorSet,
+    ) -> None:
+        """Send a fixed number of heaps.
+
+        More specifically, send `N_TX_ITEMS` Chunks, each of which contain
+        `HEAPS_PER_FENG_PER_CHUNK` heaps.
+        """
+        # Send the descriptors as the recv_stream object needs it to
+        # interpret the received heaps correctly.
+        await send_stream.stream.async_send_heap(send_stream.descriptor_heap)
+
+        for i in range(N_TX_ITEMS):
+            # Get a free chunk - there is not always a free one available. This
+            # function yields until one it available.
+            chunk = await send_stream.get_free_chunk()
+
+            # Populate the buffer with dummy data.
+            chunk.data.fill(0)
+
+            # Give the chunk back to the send_stream to transmit out
+            # onto the network.
+            chunk.timestamp = i * HEAPS_PER_FENG_PER_CHUNK * heap_timestamp_step
+            chunk.send(send_stream, time_converter, sensors)
+        # send_heap just queues data for sending but is non-blocking.
+        # Flush to ensure that the data all gets sent before we return.
+        await send_stream.stream.async_flush()
+
+    @staticmethod
+    async def _recv_data(
+        recv_stream: spead2.recv.asyncio.Stream,
+        channel_offset: int,
+        n_channels_per_substream: int,
+        n_spectra_per_heap: int,
+        heap_timestamp_step: int,
+    ) -> None:
+        """Receive data transmitted from :func:`_send_data`.
+
+        Error-check data here as well.
+        """
+        ig = spead2.ItemGroup()
+
+        # Wait for the first packet to arrive - it is expected to be the
+        # SPEAD descriptor.
+        heap = await recv_stream.get()
+        # TODO: No checks on the count sequence yet
+        items = ig.update(heap)
+        assert items == {}, "This heap contains item values not just the expected descriptors."
+
+        # Check the data heaps
+        zero_data = np.zeros(shape=(n_channels_per_substream, n_spectra_per_heap, COMPLEX), dtype=np.int8)
+        for i in range(TOTAL_HEAPS):
+            heap = await recv_stream.get()
+            items = ig.update(heap)
+            assert set(items.keys()) == {"timestamp", "frequency", "bf_raw"}
+            assert items["timestamp"].id == TIMESTAMP_ID
+            assert items["timestamp"].value == i * heap_timestamp_step
+            assert items["frequency"].id == FREQUENCY_ID
+            assert items["frequency"].value == channel_offset
+            assert items["bf_raw"].id == BF_RAW_ID
+            assert items["bf_raw"].value.shape == (n_channels_per_substream, n_spectra_per_heap, COMPLEX)
+            assert items["bf_raw"].value.dtype == np.int8
+            np.testing.assert_equal(items["bf_raw"].value, zero_data)
```

Comment:
For the purposes of this file as it currently stands, this is fine.

In general though, I'd prefer to have these assertions in the body of the test itself, rather than the helper function. That way you could re-use the helper function for other tests within the file.

No reason to change anything at this stage, but if the file expands to include more tests at some point it would be better to rearrange things.